GCTC-NTGC/gc-digital-talent
# 2834 azure yaml pipeline

Resolves #2834.

Made some updates to azure-pipeline.yml and associated scripts to get it to work.

Successfully created a pipeline in Azure DevOps which builds an artifact based on azure-pipeline.yml! It is named after our repo, since it uses the repo's yml file: "GCTC-NTGC.gc-digital-talent".

I added the Azure Pipelines Github App to our github org, with permission to access this repo. This allows Azure DevOps to connect to the repo without using one of our personal access tokens.

Successfully released the artifact to Dev server! Everything seemed to work correctly. I created a new release pipeline for this, but once this PR is accepted we should delete the new pipeline, and configure the existing pipeline to use the new artifact instead.

To test:
- Log in to Azure DevOps
- Go to Releases
- Go to the GCTalent PowerShell Push to AppSvc - YML Pipeline
- Deploy to UAT
- Manually test UAT to make sure things still work.
